### PR TITLE
Feature/hpapi 342

### PR DESCRIPTION
**Pull request by jeffbum**
This is a WIP til my timebox runs out.

### CODE REVIEW

File: src/funcs/getColumnType.spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    const response = getColumnType(columnType);
+    expect(response).toEqual('varchar')
+  })
+  it(`returns 'number' when data_type = 'double precision'.`, () => {
+    const columnType = { data_type: 'double precision'}
+    const response = getColumnType(columnType);
+    expect(response).toEqual('number')
+  })
+
+  it(`returns 'number' when data_type = 'timestamp with time zone'.`, () => {
```
Comment (khgriffi259) [inline src/funcs/getColumnType.spec.js]: should read `returns timestamp`

File: src/funcs/getColumnType.spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    const columnType = { data_type: 'timestamp with time zone'}
+    const response = getColumnType(columnType);
+    expect(response).toEqual('timestamp')
+  })
+  it(`returns 'number' when data_type = 'timestamp without time zone'.`, () => {
```
Comment (khgriffi259) [inline src/funcs/getColumnType.spec.js]: should read `returns timestamp`

File: src/funcs/getColumnType.spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const columnType = { data_type: 'timestamp without time zone'}
+    const response = getColumnType(columnType);
+    expect(response).toEqual('timestamp')
+  })
+  it(`returns 'number' when data_type = 'time without time zone'.`, () => {
```
Comment (khgriffi259) [inline src/funcs/getColumnType.spec.js]: should read `returns timestamp`

File: src/funcs/getColumnType.spec.js (hunk 1 above)
Comment (jeffbum) [reply src/funcs/getColumnType.spec.js]: done.

File: src/funcs/getColumnType.spec.js (hunk 2 above)
Comment (jeffbum) [reply src/funcs/getColumnType.spec.js]: done

File: src/funcs/getColumnType.spec.js (hunk 3 above)
Comment (jeffbum) [reply src/funcs/getColumnType.spec.js]: done

File: src/funcs/getColumnType.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+// NOTE: possible to use https://www.dbml.org/js-module/#api for the transform to dbml?
+module.exports = function getColumnType(col) {
+  const { data_type: dataType } = col;
+  let columnType;
+  switch (dataType) {
+    case 'character varying':
```
Comment (khgriffi259) [inline src/funcs/getColumnType.js:6]: should this return columnType = 'varchar'? or is 'varchar' the default value maybe?

Comment (jeffbum) [reply src/funcs/getColumnType.js:6]: it does. You can stack cases so you don't have to do multiple return lines, kind of like function overloading.

Comment (khgriffi259) [reply src/funcs/getColumnType.js:6]: oh nevermind. both character varying and varchar cases are returning 'varchar'...

**Review by khgriffi259 — APPROVED** [review]
The logic  looks sound to me from what i have read. i have approved.

File: src/funcs/getColumnSettings.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  if (primaryKeySetting) columnSettings.unshift(primaryKeySetting);
+
+  const defaultSetting = getColumnDefault(columnDefault, dataType);
+  if (defaultSetting) columnSettings.unshift(defaultSetting);
+
+  const notNullSetting = !isNullable && 'not null';
+  if (notNullSetting) columnSettings.unshift(notNullSetting);
+
+  const note = columnComment && `note: '${columnComment}'`;
+  if (note) columnSettings.unshift(note);
```
Comment (khgriffi259) [inline src/funcs/getColumnSettings.js:25]: Is the unshift an alternative to using the spread operator?

Comment (jeffbum) [reply src/funcs/getColumnSettings.js:25]: unshift moves it to the from of an array. push puts it at the end.

File: src/funcs/getColumnDefault.spec.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+const getColumnDefault = require('./getColumnDefault');
+
+describe('getColumnDefault()', () => {
+  it(`returns '' if '::' is in the default value string.` , () => {
+    const response = getColumnDefault('nextval(some.sequence.id)::regclass', 'number');
+    expect(response).toEqual('')
+  })
+  it(`returns '' around the default value` , () => {
```
Comment (camoore1979) [inline src/funcs/getColumnDefault.spec.js]:
```suggestion
  it(`returns '' (single quotes) around the default value, if type is 'varchar', 'character', 'char', 'text', or 'timestamp'` , () => {
```

File: src/funcs/getColumnDefault.spec.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    const response = getColumnDefault('some string text', 'varchar');
+    expect(response).toEqual(`default: 'some string text'`)
+  })
+  it(`does not return '' around default value` , () => {
```
Comment (camoore1979) [inline src/funcs/getColumnDefault.spec.js]:
```suggestion
  it(`does not return '' (single quotes) around default values for non-string or non-date types` , () => {
```

File: src/funcs/getColumnType.spec.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    const columnType = { data_type: 'timestamp without time zone'}
+    const response = getColumnType(columnType);
+    expect(response).toEqual('timestamp')
+  })
+  it(`returns 'timestamp' when data_type = 'time without time zone'.`, () => {
+    const columnType = { data_type: 'time without time zone'}
+    const response = getColumnType(columnType);
+    expect(response).toEqual('timestamp')
+  })
+  it(`returns undefined if not given coreect object input`, () => {
```
Comment (camoore1979) [inline src/funcs/getColumnType.spec.js]:
```suggestion
  it(`returns undefined if argument object does not have property 'data_type'`, () => {
```

File: src/funcs/getColumnType.spec.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    const columnType = { data: 'value of incorrect property'}
```
Comment (camoore1979) [inline src/funcs/getColumnType.spec.js]: make it a bit more obvious
```suggestion
    const columnType = { not_data_type: 'value of incorrect property'}
```

**Review by camoore1979 — COMMENTED** [review]
Very close. Looks good! A couple minor nits or fixes.

File: src/funcs/getColumnDefault.spec.js (hunk 6 above)
Comment (jeffbum) [reply src/funcs/getColumnDefault.spec.js]: done.

File: src/funcs/getColumnDefault.spec.js (hunk 7 above)
Comment (jeffbum) [reply src/funcs/getColumnDefault.spec.js]: done.

File: src/funcs/getColumnType.spec.js (hunk 8 above)
Comment (jeffbum) [reply src/funcs/getColumnType.spec.js]: done.

File: src/funcs/getColumnType.spec.js (hunk 9 above)
Comment (jeffbum) [reply src/funcs/getColumnType.spec.js]: done.

**Review by camoore1979 — APPROVED** [review]
👍 👍 Works locally! DBML works in dbdiagram.io. Tests run!

**Pull request merged**